@xen-orchestra/fs/src/utils.js

```diff
@@ -0,0 +1,45 @@
+import { createLogger } from '@xen-orchestra/log'
+import { TimeoutError } from 'promise-toolbox'
+const { info, warn } = createLogger('xo:fs:abstract')
+
+export function withTimeout(fn, timeout, { onSuccess, onTimeout, onSuccessAfterTimeout, onFailureAfterTimeout } = {}) {
+  if (typeof fn.then !== 'undefined') {
+    throw new Error('Function needs to be asynchronous.')
+  }
```

Comment:
I did tell something wrong here 
you can check if an object is a promis thy checking object.then. AFAIK, there is now way to know if a function return a promise before running it

Reply:
I think you should remove this code,